src/chapter01/exercises/exercise02.lean

```diff
@@ -9,13 +10,74 @@ def E : set ℝ := {1}
 -- Which pair of these sets has the property that neither is contained in the other?
 lemma parta : ∃ S T ∈ ({B, C, D, E} : set (set ℝ)), ¬ (S ⊆ T) ∧ ¬ (T ⊆ S) :=
 begin
-  sorry
+  have h0 : D = ({0} : set ℝ),
+  { simp_rw D,
+    ext,
+    split,
+    { intro hx,
+      simp at hx,
+      simp,
+      cases hx with hl hr,
+      cases hl with a ha,
+      rw ha at hr,
+      rw ha,
```

Comment:
Don't be rewriting `ha` everywhere. Use the `rcases rfl` trick to *define* `x` to be `\u a`.